src/lib/datepicker/datepicker.ts

```diff
@@ -27,6 +27,9 @@ import {MdDatepickerInput} from './datepicker-input';
 import {CalendarLocale} from '../core/datetime/calendar-locale';
 
 
+let datepickerUid = 0;
```

Comment:
Add comment about what this is? We should also probably prefix with an underscore to be consistent with other counters.